Fixes to RP2350 Framebuffer and handling of flash writes

eightycc opened the pull request:
This PR is a small package of fixes made while resolving #10104:

- Makes updates to nesting_count during interrupt disabling atomic in `common_hal_mcu_enable_interrupts()`.
- Fixes miscellaneous memory leaks and factors tear-down of partially constructed Framebuffer objects in case of resource allocation failures.
- Minor code cleanup in RP2350 Framebuffer
- Removes redundant, as of Pico SDK 2.1, XIP cache clean before flash writes and PSRAM configuration preservation around flash writes. See https://github.com/raspberrypi/pico-sdk/pull/2082 and https://github.com/raspberrypi/pico-sdk/pull/2013.
- Factors enable/disable interrupts and audio DMA pause into `supervisor_flash_pre_write()` and `supervisor_flash_post_write()`.

I'm submitting this as a draft to allow more testing.

tannewt reviewed (dismissed):
Looks good! I'm happy to merge this to make testing easier. Thanks!

eightycc commented:
Corrected an additional issue in original code with DMA command buffer allocation.

Moved to "ready for review". I've been running on RP2350 with PSRAM for a few days with no issues.

tannewt reviewed (approved):
Thank you!
